BookingsApi/BookingsApi.Domain/Hearing.cs

```diff
@@ -51,7 +51,7 @@ protected Hearing()
             ScheduledDuration = scheduledDuration;
             CaseTypeId = caseType.Id;
             HearingTypeId = hearingType?.Id;
-            HearingVenueName = hearingVenue.Name;
+            HearingVenueId = hearingVenue?.Id;
```

Comment:
a hearing venue is not optional. if it's null then let the code fail here rather than mask the issue for a later date
```suggestion
            HearingVenueId = hearingVenue.Id;
```